### PR TITLE
Fix cross_entropy label assert,test=develop

### LINKED ISSUE #2250: deeplabv3+使用develop源码编译的paddle训练报错
paddle：使用develop源码编译安装paddle;
执行指令python ./train.py --batch_size=1 --train_crop_size=769 --total_step=50 --norm_type=gn --init_weights_path=./deeplabv3plus_gn_init --save_weights_path=model --dataset_path=./data/cityscape进行训练，报错信息如下：
image
使用paddle1.4.1 deeplabv3+模型是可以训练的，安装最新的evelop源码编译安装paddle训练报该错误信息。

### PR DESCRIPTION
**Pull request by baiyfbupt**
Fix https://github.com/PaddlePaddle/models/issues/2250

### CODE REVIEW

**Comment by baiyfbupt** [comment]
@chengduoZH 
In image segmentation task, some pixels may be labeled as 255( 255 will be set as ignore_index), indicating that they are not interested. But 255 may exceed input feature_size.
So this asset may should be placed in `label != ignore_index` branch.

**Review by qingqing01 — APPROVED** [review]
LGTM.
